azure-mgmt-resources/src/main/java/com/azure/management/RestClientBuilder.java

```diff
@@ -144,6 +148,11 @@ public RestClientBuilder withConfiguration(Configuration configuration) {
         return this;
     }
 
+    public RestClientBuilder withClientId(String clientId) {

```

Comment:
Need a bit more discussion on this interface, whether need to group into withCredential?